### PR TITLE
FEATURE-IMAGE-UPLOADER

### PR DESCRIPTION
**Pull request by melas123**
REF [7](https://trello.com/c/tLz1CFZL)
## Details

```
* use FactoryGirl gem to initialize DB object for test
* Tests for model
* Tests for controller
* Write controller
```
## Others

```
    * Add pry gem
```

### CODE REVIEW

**Comment by melas123** [comment]
@SamiDghim  @medBelaid  @thameursaadi  please do a review

File: app/controllers/image_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class ImageController < ApplicationController
+  respond_to :json
+  def create
+    if params[:url] and params[:recipe_id]
+      Image.create(title: params[:title],url: params[:url],recipe: Recipe.find(params[:recipe_id]))
+    else
+      render status: 500
```
Comment (moelheni) [inline app/controllers/image_controller.rb]: remove this kind of tests. (this is the model job to verify ..)

File: spec/factories/images.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FactoryGirl.define do
+
+ 	factory :image_with_valid_url,class: Image do
```
Comment (moelheni) [inline spec/factories/images.rb:3]: Indent with 2 spaces

File: spec/models/image_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe Image, type: :model do
+  it "has a valide url and title" do
+  	expect(build(:image_with_valid_url)).to be_valid
```
Comment (moelheni) [inline spec/models/image_spec.rb:5]: Indent with 2 spaces

File: app/assets/javascripts/image.coffee [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (achr3f) [inline app/assets/javascripts/image.coffee:1]: No need to commit empty files

File: app/helpers/image_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module ImageHelper
```
Comment (achr3f) [inline app/helpers/image_helper.rb:1]: you can delete this file

File: app/controllers/image_controller.rb (hunk 1 above)
Comment (melas123) [inline app/controllers/image_controller.rb]: done

File: app/controllers/image_controller.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        render status: 500
+      end
+    else
+      render status: 500
+    end
+  end
+
+  def update
+    if params[:id] and params[:title]
+      if Image.exists?(params[:id])
```
Comment (moelheni) [inline app/controllers/image_controller.rb]: no need for this conditions

File: spec/controllers/image_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      get :index,format: :json,recipe_id: recipe.id
+      expect(assigns[:images].length).to eq(10)
+    end
+  end
+
+  describe "GET delete" do
+  let!(:image_with_valid_url) { create(:image_with_valid_url) }
+
+    it "deletes image when it gets an image id" do
+      #image = create( :image_with_valid_url)
```
Comment (SamiDghim) [inline spec/controllers/image_controller_spec.rb]: delete this comment !!

**Pull request merged**